### PR TITLE
[CLI] BOOST-535: deal with lambda character limit

### PR DESCRIPTION
**Pull request by MYF95**
## Description
Previous character limit of project name was on 23 characters. Changed the suffix `-application-stack` to `-app` to increase the overall limit to 37 characters.

## Changes
* Updated the CLI with actual project name length
* Added tests for `assertNameIsCorrect`
* Changed resource names using `-application-stack` to `-app`

## Checks
- [X] Project Builds
- [X] Project passes tests and checks
- [ ] Updated documentation accordingly at [the docs repo](https://github.com/boostercloud/docs)
- [ ] Added a link pointing to the documentation PR in this PR

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "@types/inquirer": "^6.5.0",
     "@types/mustache": "^0.8.32",
     "@types/rewire": "^2.5.28",
+    "faker": "^4.1.0",
```
Comment (glammers1) [inline packages/cli/package.json:40]: I would say that `faker` was already accessible for the cli package.

Comment (MYF95) [reply packages/cli/package.json:40]: I wasn't able to import the package from cli without installing it

Comment (MYF95) [reply packages/cli/package.json]: Removed the package and worked perfectly, updated the package.json, thanks @glammers1

**Comment by MYF95** [comment]
bot: integration

**Comment by glammers1** [comment]
bot: integration

File: packages/cli/src/services/provider-service.ts [hunk 2]
```diff
@@ -2,10 +2,12 @@ import { BoosterConfig } from '@boostercloud/framework-types'
...
-  // It is 55 because cloudformations max length is 63.
-  // Booster creates an S3 bucket ended in '-toolkit'
-  // which is 8 chars long. 63 - 8 = 55
-  const maxProjectNameLength = 55
+  // Current characters max length: 37
+  // Lambda name limit is 64 characters
+  // `-subscriptions-notifier` lambda is 23 characters
+  // `-app` prefix is added to application stack
+  // which is 64 - 23 - 4 = 37
+  const maxProjectNameLength = 37
```
[the remark this answers is not in the document]
Comment (alvaroloes) [reply packages/cli/src/services/provider-service.ts:10]: Good suggestion! Each provider will have its own limitations.

@MYF95 Could you please create a Jira for this?

File: packages/cli/test/services/provider-service.test.ts [hunk 3]
```diff
@@ -1,6 +1,43 @@
...
+import { random } from 'faker'
 
 describe('providerService', () => {
+  afterEach(() => {
+    restore()
+  })
+
+  describe('assertNameIsCorrect', () => {
+    it('should throw an error on surpassing project name max length', () => {
+      const inputString = random.alphaNumeric(38)
```
Comment (tainguyenbui) [inline packages/cli/test/services/provider-service.test.ts:15]: even a better one

```suggestion
      const inputString = random.alphaNumeric(random.number( { min: 38 } ))
```

File: packages/cli/src/services/provider-service.ts (hunk 2 above)
Comment (tainguyenbui) [inline packages/cli/src/services/provider-service.ts:10]:
```suggestion
  const MAX_PROJECT_NAME_LENGTH = 37
```

Comment (MYF95) [reply packages/cli/src/services/provider-service.ts:10]: I think we are using camelCase throughout the project for constants, should we start changing that?

**Comment by glammers1** [comment]
@MYF95 please merge/rebase this branch with master so that we can execute the integration tests since we already merged #161

**Comment by MYF95** [comment]
bot: integration

**Comment by MYF95** [comment]
bot: integration

**Pull request merged**